### PR TITLE
Add doxygen/1.8.17 recipe

### PR DESCRIPTION
**Pull request by robert-shade**
Specify library name and version:  doxygen/1.8.17

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/doxygen/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    def requirements(self):
+        self.requires("libiconv/1.15")
+
+    def build_requirements(self):
+        if not tools.which("flex"):
+            self.build_requires("flex/2.6.4@bincrafters/stable")
+
+        if not tools.which("bison"):
+            self.build_requires("bison/3.3.2@bincrafters/stable")
```
Comment (gocarlos) [inline recipes/doxygen/all/conanfile.py]: only CCI packages allowed here, so we need to port `flex` and `bison` first to CCI

Comment (gocarlos) [reply recipes/doxygen/all/conanfile.py]: e.g. from bincrafters to this repo

**Comment by Croydon** [comment]
If you are building doxygen from source with all features enabled you need many dependencies which we don't have 

https://github.com/conan-io/wishlist/issues/115

At Bincrafters we have one which is re-using official binaries, but looks pretty ugly. I will probably migrate this to CCI at some point and we can switch to building from source when we have all the dependencies

https://github.com/bincrafters/conan-doxygen_installer/

File: recipes/doxygen/all/conanfile.py (hunk 1 above)
Comment (madebr) [inline recipes/doxygen/all/conanfile.py:24]: This is available
```suggestion
            self.build_requires("bison/3.5.3")
```

File: recipes/doxygen/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        self.requires("libiconv/1.15")
+
+    def build_requirements(self):
+        if not tools.which("flex"):
+            self.build_requires("flex/2.6.4@bincrafters/stable")
```
Comment (madebr) [inline recipes/doxygen/all/conanfile.py:21]: Still on review at #1195
```suggestion
            self.build_requires("flex/2.6.4")
```

**Comment by Croydon** [comment]
Until we have ALL dependencies (see e.g. https://github.com/conan-io/wishlist/issues/115) we should not build it from source, as we probably miss out on several optional Doxygen features.

https://github.com/conan-io/conan-center-index/pull/1702 is merged now which re-packages official Doxygen binaries. And I'm going now to deprecate the [Bincrafters' recipe](https://github.com/bincrafters/conan-doxygen_installer) which does the same.

I would recommend to close this pull request for now. Let's have a look into building it from source again when we have all dependencies in CCI.

**Pull request closed without merging**